java-frontend/src/main/java/org/sonar/java/cfg/LiveVariables.java

```diff
@@ -47,18 +50,24 @@
 
   private final CFG cfg;
   private final Map<CFG.Block, Set<Symbol>> out = new HashMap<>();
+  private final Map<CFG.Block, Set<Symbol>> in = new HashMap<>();
+  private final boolean includeFields;
 
-  private LiveVariables(CFG cfg) {
+  private LiveVariables(CFG cfg, boolean includeFields) {
     this.cfg = cfg;
+    this.includeFields = includeFields;
   }
 
   public Set<Symbol> getOut(CFG.Block block) {
     return out.get(block);
   }
 
-  public static LiveVariables analyze(CFG cfg) {
-    LiveVariables liveVariables = new LiveVariables(cfg);
-    final Map<CFG.Block, Set<Symbol>> in = new HashMap<>();
+  public Set<Symbol> getIn(CFG.Block block) {
+    return in.get(block);
+  }
+
+  public static LiveVariables analyze(CFG cfg, boolean includeFields) {
```

Comment:
Design wise : I would have two public static methods `analyze` and `analyzeWithFields` that would call this method privately with the flag set correctly.